File: app/src/main/java/com/jhworks/imageselect/MainActivity.java
package com.jhworks.imageselect;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.TextView;

import com.jhworks.library.ImageSelector;
import com.jhworks.library.bean.MediaSelectConfig;

import java.util.ArrayList;


public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_IMAGE = 2;

    private TextView mResultText;
    private RadioGroup mChoiceMode, mShowCamera;
    private EditText mRequestNum;
    private EditText mImageSpanCount;
    private Button mBtnOpenCameraOnly;
    private Toolbar mToolbar;

    private ArrayList<String> mSelectPath;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);

        mToolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(mToolbar);
        mToolbar.setTitle(R.string.app_name);
        mToolbar.setNavigationIcon(R.mipmap.ic_launcher);

        mResultText = (TextView) findViewById(R.id.result);
        mChoiceMode = (RadioGroup) findViewById(R.id.choice_mode);
        mShowCamera = (RadioGroup) findViewById(R.id.show_camera);
        mRequestNum = (EditText) findViewById(R.id.request_num);
        mImageSpanCount = (EditText) findViewById(R.id.image_span_count);
        mBtnOpenCameraOnly = (Button) findViewById(R.id.btn_open_camera_only);

        mChoiceMode.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup radioGroup, int checkedId) {
                if (checkedId == R.id.multi) {
                    mRequestNum.setEnabled(true);
                } else {
                    mRequestNum.setEnabled(false);
                    mRequestNum.setText("");
                }
            }
        });

        View button = findViewById(R.id.button);
        if (button != null) {
            button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    pickImage(false);
                }
            });
        }

        mBtnOpenCameraOnly.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                pickImage(true);
            }
        });

        findViewById(R.id.btn_open_video).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                pickVideo();
            }
        });
    }

    private void pickVideo() {
        int maxNum = 9;

        if (!TextUtils.isEmpty(mRequestNum.getText())) {
            try {
                maxNum = Integer.valueOf(mRequestNum.getText().toString());
            } catch (NumberFormatException e) {
                e.printStackTrace();
            }
        }
        int imageSpanCount = 3;
        if (!TextUtils.isEmpty(mImageSpanCount.getText())) {
            try {
                imageSpanCount = Integer.valueOf(mImageSpanCount.getText().toString());
            } catch (NumberFormatException e) {
                e.printStackTrace();
            }
        }
        if (mSelectPath != null && mSelectPath.size() > 0) mSelectPath.clear();
        MediaSelectConfig config = new MediaSelectConfig()
                .setSelectMode(mChoiceMode.getCheckedRadioButtonId() == R.id.single ?
                        MediaSelectConfig.MODE_SINGLE : MediaSelectConfig.MODE_MULTI)
                .setOriginData(mSelectPath)
                .setMaxCount(maxNum)
                .setImageSpanCount(imageSpanCount);

        ImageSelector.create()
                .setMediaConfig(config)
                .startVideoAction(MainActivity.this, REQUEST_IMAGE);
    }

    private void pickImage(boolean isOpneCameraOnly) {
        boolean showCamera = mShowCamera.getCheckedRadioButtonId() == R.id.show;
        int maxNum = 9;

        if (!TextUtils.isEmpty(mRequestNum.getText())) {
            try {
                maxNum = Integer.valueOf(mRequestNum.getText().toString());
            } catch (NumberFormatException e) {
                e.printStackTrace();
            }
        }
        int imageSpanCount = 3;
        if (!TextUtils.isEmpty(mImageSpanCount.getText())) {
            try {
                imageSpanCount = Integer.valueOf(mImageSpanCount.getText().toString());
            } catch (NumberFormatException e) {
                e.printStackTrace();
            }
        }
        if (mSelectPath != null && mSelectPath.size() > 0) mSelectPath.clear();

        MediaSelectConfig config = new MediaSelectConfig()
                .setSelectMode(mChoiceMode.getCheckedRadioButtonId() == R.id.single ?
                        MediaSelectConfig.MODE_SINGLE : MediaSelectConfig.MODE_MULTI)
                .setOriginData(mSelectPath)
                .setShowCamera(showCamera)
                .setOpenCameraOnly(isOpneCameraOnly)
                .setMaxCount(maxNum)
                .setImageSpanCount(imageSpanCount);

        ImageSelector.create()
                .setMediaConfig(config)
                .startImageAction(MainActivity.this, REQUEST_IMAGE);

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_IMAGE) {
            if (resultCode == RESULT_OK) {
                mSelectPath = data.getStringArrayListExtra(ImageSelector.EXTRA_RESULT);
                StringBuilder sb = new StringBuilder();
                for (String p : mSelectPath) {
                    sb.append(p);
                    sb.append("\n");
                }
                mResultText.setText(sb.toString());
            }
        }
    }
}

File: app/src/main/java/com/jhworks/imageselect/MainActivity.kt
package com.jhworks.imageselect

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.RadioGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.jhworks.library.ImageSelector
import com.jhworks.library.core.MediaSelectConfig
import com.jhworks.library.core.vo.SelectMode
import java.util.*

class MainActivity : AppCompatActivity() {
    private var mResultText: TextView? = null
    private var mChoiceMode: RadioGroup? = null
    private var mShowCamera: RadioGroup? = null
    private lateinit var mRgTheme: RadioGroup
    private var mRequestNum: EditText? = null
    private var mImageSpanCount: EditText? = null
    private var mBtnOpenCameraOnly: Button? = null
    private var mToolbar: Toolbar? = null
    private var mSelectPath: ArrayList<String>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        mToolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(mToolbar)
        mToolbar!!.setTitle(R.string.app_name)
        mToolbar!!.setNavigationIcon(R.mipmap.ic_launcher)
        mResultText = findViewById<View>(R.id.result) as TextView
        mChoiceMode = findViewById<View>(R.id.choice_mode) as RadioGroup
        mShowCamera = findViewById<View>(R.id.show_camera) as RadioGroup
        mRgTheme = findViewById(R.id.rg_theme)
        mRequestNum = findViewById<View>(R.id.request_num) as EditText
        mImageSpanCount = findViewById<View>(R.id.image_span_count) as EditText
        mBtnOpenCameraOnly = findViewById<View>(R.id.btn_open_camera_only) as Button
        mChoiceMode!!.setOnCheckedChangeListener { _, checkedId ->
            if (checkedId == R.id.multi) {
                mRequestNum!!.isEnabled = true
            } else {
                mRequestNum!!.isEnabled = false
                mRequestNum!!.setText("")
            }
        }
        val button = findViewById<View>(R.id.button)
        button?.setOnClickListener { pickImage(false) }
        mBtnOpenCameraOnly!!.setOnClickListener { pickImage(true) }
        findViewById<View>(R.id.btn_open_video).setOnClickListener { pickVideo() }

        findViewById<View>(R.id.btn_test_crop).setOnClickListener {
            startActivity(Intent(this, TestImageCropActivity::class.java))
        }
    }

    private fun pickVideo() {
        var maxNum = 9
        if (!TextUtils.isEmpty(mRequestNum!!.text)) {
            try {
                maxNum = Integer.valueOf(mRequestNum!!.text.toString())
            } catch (e: NumberFormatException) {
                e.printStackTrace()
            }
        }
        var imageSpanCount = 3
        if (!TextUtils.isEmpty(mImageSpanCount!!.text)) {
            try {
                imageSpanCount = Integer.valueOf(mImageSpanCount!!.text.toString())
            } catch (e: NumberFormatException) {
                e.printStackTrace()
            }
        }
        if (mSelectPath != null && mSelectPath!!.size > 0) mSelectPath!!.clear()

        ImageSelector.startVideoAction(this@MainActivity, REQUEST_IMAGE,
                MediaSelectConfig.Builder()
                        .setSelectMode(getSelectMode())
                        .setOriginData(mSelectPath)
                        .setMaxCount(maxNum)
                        .setImageSpanCount(imageSpanCount)
                        .build())
    }

    private fun pickImage(isOpneCameraOnly: Boolean) {
        val showCamera = mShowCamera!!.checkedRadioButtonId == R.id.show
        var maxNum = 9
        if (!TextUtils.isEmpty(mRequestNum!!.text)) {
            try {
                maxNum = Integer.valueOf(mRequestNum!!.text.toString())
            } catch (e: NumberFormatException) {
                e.printStackTrace()
            }
        }
        var imageSpanCount = 3
        if (!TextUtils.isEmpty(mImageSpanCount!!.text)) {
            try {
                imageSpanCount = Integer.valueOf(mImageSpanCount!!.text.toString())
            } catch (e: NumberFormatException) {
                e.printStackTrace()
            }
        }
//        if (mSelectPath != null && mSelectPath!!.size > 0) mSelectPath!!.clear()

        // theme
        val theme = when (mRgTheme.checkedRadioButtonId) {
            R.id.theme_custom -> R.style.sl_theme_custom
            R.id.theme_light -> R.style.sl_theme_light
            R.id.theme_dark -> R.style.sl_theme_dark
            else -> R.style.sl_theme_light
        }

        ImageSelector.startImageAction(this@MainActivity, REQUEST_IMAGE,
                MediaSelectConfig.Builder()
                        .setSelectMode(getSelectMode())
                        .setOriginData(mSelectPath)
                        .setTheme(theme)
                        .setShowCamera(showCamera)
                        .setPlaceholderResId(R.mipmap.ic_launcher)
                        .setOpenCameraOnly(isOpneCameraOnly)
                        .setMaxCount(maxNum)
                        .setImageSpanCount(imageSpanCount)
                        .build()
        )
    }

    private fun getSelectMode(): Int {
        return if (mChoiceMode!!.checkedRadioButtonId == R.id.single)
            SelectMode.MODE_SINGLE
        else
            SelectMode.MODE_MULTI
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_IMAGE) {
            if (resultCode == Activity.RESULT_OK) {
                mSelectPath = ImageSelector.getSelectResults(data)
                val sb = StringBuilder()
                mSelectPath?.forEach {
                    sb.append(it)
                    sb.append("\n")
                }
                mResultText?.text = sb.toString()
                Log.e("ImageSelect", "结果： $mSelectPath")
            }
        }
    }

    companion object {
        private const val REQUEST_IMAGE = 2
    }
}